VideoLocker/src/main/java/org/edx/mobile/view/DiscussionAddPostFragment.java

```diff
@@ -215,35 +221,91 @@ protected void getTopicList() {
-                ArrayAdapter<String> adapter = new ArrayAdapter<String>(container.getContext(), R.layout.edx_spinner_item, topics) {
+                final int basePadding = getResources().getDimensionPixelSize(R.dimen.widget_margin);
+                final int extraPaddingPerLevel = getResources().getDimensionPixelSize(R.dimen.edx_margin);
+                final ArrayAdapter<DiscussionTopicDepth> adapter = new ArrayAdapter<DiscussionTopicDepth>(container.getContext(), 0, allTopicsWithDepth) {
                     @Override
                     public View getView(int position, View convertView, ViewGroup parent) {
-                        final TextView view = (TextView) super.getView(position, convertView, parent);
-                        view.setText(prefix + view.getText().toString());
+                        final TextView view;
+                        if (convertView == null) {
+                            view = (TextView) LayoutInflater.from(parent.getContext()).inflate(R.layout.edx_spinner_item, parent, false);
+                        } else {
+                            view = (TextView) convertView;
+                        }
+                        final DiscussionTopicDepth topic = getItem(position);
+                        final String text;
+                        if (null == topic) {
+                            text = getString(R.string.discussion_add_post_choose_a_topic);
+                        } else {
+                            text = getString(R.string.discussion_add_post_topic_label) + ": " + topic.getDiscussionTopic().getName();
+                        }
+                        view.setText(text);
+                        return view;
+                    }
+
+                    @Override
+                    public View getDropDownView(int position, View convertView, ViewGroup parent) {
+                        final TextView view;
+                        if (convertView == null) {
+                            view = (TextView) LayoutInflater.from(parent.getContext()).inflate(R.layout.edx_spinner_dropdown_item, parent, false);
+                        } else {
+                            view = (TextView) convertView;
+                        }
+                        final DiscussionTopicDepth topic = getItem(position);
+                        final int extraLeftPadding;
+                        final String text;
+                        if (null == topic) {
+                            extraLeftPadding = 0;
+                            text = getString(R.string.discussion_add_post_choose_a_topic);
+                        } else {
+                            extraLeftPadding = topic.getDepth() * extraPaddingPerLevel;
+                            text = topic.getDiscussionTopic().getName();
+                        }
+                        view.setPaddingRelative(basePadding + extraLeftPadding, view.getPaddingTop(), basePadding, view.getPaddingBottom());
```

Comment:
The `setPaddingRelative()` method was only introduced in API 17. We should probably use the `ViewCompat` utility to do this in an API-safe manner.